packages/core/test/merkle-distributor/MerkleDistributor.js

```diff
@@ -315,166 +304,283 @@ contract("MerkleDistributor.js", function(accounts) {
+          assert.isFalse(await merkleDistributor.verifyClaim(claim));
+        };
+        // Incorrect account:
+        await isInvalidProof({
           windowIndex: windowIndex,
-          account: leaf.account,
+          account: rando,
+          accountIndex: leaf.accountIndex,
           amount: leaf.amount,
           merkleProof: claimerProof
         });
-        // Compare gas used against benchmark implementation: Uniswap's "single window" Merkle distributor,
-        // that uses a Bitmap instead of mapping between addresses and booleans to track claims.
-        assert.equal(claimTx.receipt.gasUsed, 92029);
-      });
-      it("invalid proof", async function() {
-        // Incorrect account:
-        assert(
-          await didContractThrow(
-            merkleDistributor.claimWindow({
-              windowIndex: windowIndex,
-              account: rando,
-              amount: leaf.amount,
-              merkleProof: claimerProof
-            })
-          )
-        );
 
         // Incorrect amount:
         const invalidAmount = "1";
-        assert(
-          await didContractThrow(
-            merkleDistributor.claimWindow({
-              windowIndex: windowIndex,
-              account: leaf.account,
-              amount: invalidAmount,
-              merkleProof: claimerProof
-            })
-          )
-        );
+        await isInvalidProof({
+          windowIndex: windowIndex,
+          account: leaf.account,
+          accountIndex: leaf.accountIndex,
+          amount: invalidAmount,
+          merkleProof: claimerProof
+        });
+
+        // Incorrect account index:
+        const invalidAccountIndex = "99";
+        await isInvalidProof({
+          windowIndex: windowIndex,
+          account: leaf.account,
+          accountIndex: invalidAccountIndex,
+          amount: leaf.amount,
+          merkleProof: claimerProof
+        });
 
         // Invalid merkle proof:
         const invalidProof = [utf8ToHex("0x")];
-        assert(
-          await didContractThrow(
-            merkleDistributor.claimWindow({
-              windowIndex: windowIndex,
-              account: leaf.account,
-              amount: leaf.amount,
-              merkleProof: invalidProof
-            })
-          )
-        );
+        await isInvalidProof({
+          windowIndex: windowIndex,
+          account: leaf.account,
+          accountIndex: leaf.accountIndex,
+          amount: leaf.amount,
+          merkleProof: invalidProof
+        });
       });
     });
-  });
-  describe("(claimWindows)", function() {
-    let rewardRecipients1, rewardRecipients2;
-    let rewardLeafs1, rewardLeafs2;
-    let merkleTree1, merkleTree2;
-    beforeEach(async function() {
-      // Assume we start at first windowIndex. Disable vesting.
-      windowIndex = 0;
-      const currentTime = await timer.getCurrentTime();
-      windowStart = currentTime;
+    describe("(claimWindows)", function() {
```

Comment:
can we also add another test where we have claim windows with different erc20 token payouts, and then a batch claim where a user submits claim across multiple tokens?

Reply:
Yep

Reply:
Done